lib/internal/Magento/Framework/Mail/TransportInterfaceFactory.php

```diff
@@ -49,4 +49,14 @@ public function create(array $data = [])
     {
         return $this->_objectManager->create($this->_instanceName, $data);
     }
+
+    /**
+     * @param $name
+     * @return $this
+     */
+    public function setInstanceName($name)
```

Comment:
Agree with Rick. Making such call from plugins doesn't make it non-conflicting. Preferences are a standard declarative way of doing conflicting modifications. It's easier to track them to find conflicts between modules.